sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/worker/IsmFormat.java

```diff
@@ -840,59 +745,29 @@ protected long getEncodedElementByteSize(KeyPrefix value, Coder.Context context)
    *   <li>0x01 (version key as a single byte)</li>
    * </ul>
    */
-  static class Footer {
+  @AutoValue
+  abstract static class Footer {
     static final int LONG_BYTES = 8;
     static final int FIXED_LENGTH = 3 * LONG_BYTES + 1;
     static final byte VERSION = 2;
 
-    private final long indexPosition;
-    private final long bloomFilterPosition;
-    private final long numberOfKeys;
-
-    Footer(long indexPosition, long bloomFilterPosition, long numberOfKeys) {
-      this.indexPosition = indexPosition;
-      this.bloomFilterPosition = bloomFilterPosition;
-      this.numberOfKeys = numberOfKeys;
-    }
-
-    public long getIndexPosition() {
-      return indexPosition;
-    }
-
-    public long getBloomFilterPosition() {
-      return bloomFilterPosition;
-    }
-
-    public long getNumberOfKeys() {
-      return numberOfKeys;
-    }
+    public abstract long getIndexPosition();
```

Comment:
There's no magic mapping in IsmShard/IsmRecord. AutoValue will use abstract properties with either the bean get/is* prefix (getIndexPosition()), or without prefix (prefixPosition()), but should be consistent within a type.